RecursiveExtractor.Tests/ExtractorTests.cs

```diff
@@ -41,6 +42,64 @@ public void ExtractArchive(string fileName, int expectedNumFiles = 26)
             Assert.IsTrue(results.Count() == expectedNumFiles);
         }
 
+        [DataTestMethod]
+        [DataRow("SharedEncrypted.zip")]
+        [DataRow("SharedEncrypted.7z")]
+        [DataRow("SharedEncrypted.rar4")]
+        // [DataRow("SharedEncrypted.rar")] // RAR5 is not yet supported by SharpCompress: https://github.com/adamhathcock/sharpcompress/issues/517
+        public void ExtractEncryptedArchive(string fileName, int expectedNumFiles = 26)
+        {
+            var extractor = new Extractor();
+            var path = Path.Combine(Directory.GetCurrentDirectory(), "TestData", fileName);
+            var results = extractor.ExtractFile(path, new ExtractorOptions()
+            {
+                Passwords = new Dictionary<Regex, List<string>>()
+                {
+                    {
+                        new Regex(".*"), 
+                        new List<string>()
+                        {
+                            "AnIncorrectPassword",
+                            "TheMagicWordIsPotato"
+                        } 
+                    } 
+                }
+            }).ToList();
```

Comment:
Nit on need/value of calling .ToList() when ExtractFile already returns IEnumerable and other examples do not bother.

Reply:
Fixed.